### ISSUE TITLE
fails to load the profiled module without specifying sys.path

### ISSUE THREAD

**Issue by lucsorel** [joerick/pyinstrument#112]
First, thank you for maintaining this great project :)

However, I had a hard time using pyinstrument to profile a module. I carefully read PR https://github.com/joerick/pyinstrument/pull/45 which enabled profiling a module but it did not help me writing the proper command. I managed to make it work but I am not sure whether the workaround I came up with is good:

- if it is not, I am interested in knowing the proper approach
- if it is, I believe it should be documented in order to help the user community (I would be glad to provide a PR for that in this case)

Here is my use case:

- I have a `profile_me.py` script (in a 'test' folder), which I would like to profile as a module:

```python
from random import choice
from time import sleep
from sys import path

def i_do_things():
    for x in range(5):
        sleep(choice(range(1,5))/100)

if __name__ == '__main__':
    print('sys.path:', path)
    i_do_things()
```

- the following 'script' profiling command works:

```sh
pyinstrument profile_me.py
# ouputs (I have shortened the paths list, which does not contain the 'test' folder holding profile_me.py):
sys.path: ['', '...test/.venv/bin', '.../3.8.6/lib/python38.zip', '.../3.8.6/lib/python3.8', '.../3.8.6/lib/python3.8/lib-dynload', '...test/.venv/lib/python3.8/site-packages']
[...]
Program: profile_me.py
0.142 <module>  profile_me.py:1
└─ 0.142 i_do_things  profile_me.py:4
   └─ 0.142 sleep  <built-in>:0
```

- the following 'module' profiling command fails:
```sh
pyinstrument -m profile_me
```

The traceback is:

```sh
Traceback (most recent call last):
  File "[...]/.venv/bin/pyinstrument", line 8, in <module>
    sys.exit(main())
  File "[...]/.venv/lib/python3.8/site-packages/pyinstrument/__main__.py", line 137, in main
    exec_(code, globs, None)
  File "<string>", line 1, in <module>
  File "[...]/3.8.6/lib/python3.8/runpy.py", line 203, in run_module
    mod_name, mod_spec, code = _get_module_details(mod_name)
  File "[...]/3.8.6/lib/python3.8/runpy.py", line 138, in _get_module_details
    raise error("No module named %s" % mod_name)
ImportError: No module named profile_me
```

Runpy actually fails to load my module because it does not know where to search it.

By setting the PYTHONPATH environment variable to the directory containing the script (used to feed directories to sys.path), I managed to make it work:

```sh
# pwd outputs the current working directory where the command is launched. Here, it is the folder containing profile_me.py
export PYTHONPATH=$(pwd) && poetry run pyinstrument -m profile_me
# outputs (this time, the paths list contains the 'test' folder holding profile_me.py):
sys.path: ['.../test/.venv/bin', '.../test', '.../3.8.6/lib/python38.zip', '.../3.8.6/lib/python3.8', '.../3.8.6/lib/python3.8/lib-dynload', '.../test/.venv/lib/python3.8/site-packages']
[...]
Program: profile_me
0.122 <module>  <string>:1
└─ 0.122 run_module  runpy.py:197
      [2 frames hidden]  runpy
         0.122 _run_code  runpy.py:64
         └─ 0.122 <module>  profile_me.py:1
            └─ 0.122 i_do_things  profile_me.py:4
               └─ 0.122 sleep  <built-in>:0
```

I wonder whether it is a good practice to hack the PYTHONPATH environment variable (otherwise undefined) or whether this workaround should be documented?

**Comment by joerick** [comment]
Hmm. Could you check a couple things for me? Does `python -m profile_me` work as you expect? How about `python -m profile -m profile_me` and `python -m cProfile -m profile_me`?

**Comment by lucsorel** [comment]
Hi @joerick.

Thank you for giving consideration to this topic.

The script does not do much, but at least it prints the sys.path out. Here are the checks you asked me to perform (I ensured that `PYTHONPATH` was undefined each time).

* `python -m profile_me`: works as expected (the current directory is fed into sys.path):
```sh
python -m profile_me
# outputs:
sys.path: ['...test/', '.../3.8.6/lib/python38.zip', '.../3.8.6/lib/python3.8', '.../3.8.6/lib/python3.8/lib-dynload', '.../3.8.6/lib/python3.8/site-packages']
```

* I am not familiar with profiling, it is a new topic for me. I am thus unsure about what are the expected outcome of `python -m profile -m profile_me` (pure python profiler). But the following outputs make some sense to me (at least, nothing crashes):

```sh
python -m profile -m profile_me
# outputs:
sys.path: ['...test', '/home/luc/.pyenv/versions/3.8.6/lib/python38.zip', '/home/luc/.pyenv/versions/3.8.6/lib/python3.8', '/home/luc/.pyenv/versions/3.8.6/lib/python3.8/lib-dynload', '/home/luc/.pyenv/versions/3.8.6/lib/python3.8/site-packages']
         1268 function calls (1240 primitive calls) in 0.007 seconds

   Ordered by: standard name

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        2    0.000    0.000    0.000    0.000 :0(__build_class__)
        3    0.000    0.000    0.000    0.000 :0(_fix_co_filename)
       33    0.000    0.000    0.000    0.000 :0(acquire_lock)
       12    0.000    0.000    0.000    0.000 :0(allocate_lock)
        5    0.000    0.000    0.000    0.000 :0(bit_length)
        4    0.000    0.000    0.000    0.000 :0(create_dynamic)
       11    0.000    0.000    0.000    0.000 :0(endswith)
      4/1    0.000    0.000    0.005    0.005 :0(exec)
        4    0.000    0.000    0.000    0.000 :0(exec_dynamic)
        1    0.000    0.000    0.000    0.000 :0(exp)
        9    0.000    0.000    0.000    0.000 :0(from_bytes)
       13    0.000    0.000    0.000    0.000 :0(fspath)
       12    0.000    0.000    0.000    0.000 :0(get)
       12    0.000    0.000    0.000    0.000 :0(get_ident)
       36    0.000    0.000    0.000    0.000 :0(getattr)
       14    0.000    0.000    0.000    0.000 :0(getrandbits)
       30    0.000    0.000    0.000    0.000 :0(hasattr)
        7    0.000    0.000    0.000    0.000 :0(is_builtin)
        7    0.000    0.000    0.000    0.000 :0(is_frozen)
       30    0.000    0.000    0.000    0.000 :0(isinstance)
       78    0.000    0.000    0.000    0.000 :0(join)
       23    0.000    0.000    0.000    0.000 :0(len)
        3    0.000    0.000    0.000    0.000 :0(loads)
        2    0.000    0.000    0.000    0.000 :0(log)
        3    0.000    0.000    0.000    0.000 :0(open_code)
        6    0.000    0.000    0.000    0.000 :0(pop)
        1    0.000    0.000    0.000    0.000 :0(print)
        3    0.000    0.000    0.000    0.000 :0(read)
        1    0.000    0.000    0.000    0.000 :0(register_at_fork)
       33    0.000    0.000    0.000    0.000 :0(release_lock)
       44    0.000    0.000    0.000    0.000 :0(rpartition)
      150    0.000    0.000    0.000    0.000 :0(rstrip)
        1    0.000    0.000    0.000    0.000 :0(seed)
        1    0.002    0.002    0.002    0.002 :0(setprofile)
        5    0.000    0.000    0.000    0.000 :0(sleep)
        1    0.000    0.000    0.000    0.000 :0(sqrt)
        2    0.000    0.000    0.000    0.000 :0(startswith)
       27    0.000    0.000    0.000    0.000 :0(stat)
        1    0.000    0.000    0.000    0.000 :0(update)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:103(release)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:143(__init__)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:147(__enter__)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:151(__exit__)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:157(_get_module_lock)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:176(cb)
     10/1    0.000    0.000    0.003    0.003 <frozen importlib._bootstrap>:211(_call_with_frames_removed)
       86    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:222(_verbose_message)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:342(__init__)
        2    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:35(_new_module)
        9    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:376(cached)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:389(parent)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:397(has_location)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:477(_init_module_attrs)
        6    0.000    0.000    0.001    0.000 <frozen importlib._bootstrap>:549(module_from_spec)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:58(__init__)
      6/1    0.000    0.000    0.003    0.003 <frozen importlib._bootstrap>:650(_load_unlocked)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:725(find_spec)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:78(acquire)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:800(find_spec)
       21    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:863(__enter__)
       21    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:867(__exit__)
        7    0.000    0.000    0.002    0.000 <frozen importlib._bootstrap>:890(_find_spec)
      6/1    0.000    0.000    0.004    0.004 <frozen importlib._bootstrap>:956(_find_and_load_unlocked)
      6/1    0.000    0.000    0.004    0.004 <frozen importlib._bootstrap>:986(_find_and_load)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1010(path_stats)
        4    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1088(__init__)
        4    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1099(create_module)
        4    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1107(exec_module)
       23    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1265(_path_importer_cache)
        7    0.000    0.000    0.002    0.000 <frozen importlib._bootstrap_external>:1302(_get_spec)
        7    0.000    0.000    0.002    0.000 <frozen importlib._bootstrap_external>:1334(find_spec)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1426(_get_spec)
       17    0.000    0.000    0.001    0.000 <frozen importlib._bootstrap_external>:1431(find_spec)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:294(cache_from_source)
       17    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:40(_relax_case)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:424(_get_cached)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:456(_check_name_wrapper)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:493(_classify_pyc)
        9    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:51(_unpack_uint32)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:526(_validate_timestamp_pyc)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:578(_compile_bytecode)
       72    0.000    0.000    0.001    0.000 <frozen importlib._bootstrap_external>:62(_path_join)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:629(spec_from_file_location)
       72    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:64(<listcomp>)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:68(_path_split)
        2    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:774(create_module)
      2/1    0.000    0.000    0.003    0.003 <frozen importlib._bootstrap_external>:777(exec_module)
       27    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:80(_path_stat)
        3    0.000    0.000    0.001    0.000 <frozen importlib._bootstrap_external>:849(get_code)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:90(_path_is_mode_type)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:939(__init__)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:964(get_filename)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:969(get_data)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:99(_path_isfile)
        1    0.000    0.000    0.005    0.005 <string>:1(<module>)
        1    0.000    0.000    0.000    0.000 bisect.py:1(<module>)
        0    0.000             0.000          profile:0(profiler)
        1    0.000    0.000    0.007    0.007 profile:0(run_module(modname, run_name='__main__'))
        1    0.000    0.000    0.004    0.004 profile_me.py:1(<module>)
        1    0.000    0.000    0.001    0.001 profile_me.py:5(i_do_things)
        1    0.000    0.000    0.003    0.003 random.py:1(<module>)
        1    0.000    0.000    0.000    0.000 random.py:103(__init_subclass__)
        1    0.000    0.000    0.000    0.000 random.py:123(seed)
        5    0.000    0.000    0.000    0.000 random.py:250(_randbelow_with_getrandbits)
        5    0.000    0.000    0.000    0.000 random.py:285(choice)
        1    0.000    0.000    0.000    0.000 random.py:709(SystemRandom)
        1    0.000    0.000    0.000    0.000 random.py:78(Random)
        1    0.000    0.000    0.000    0.000 random.py:94(__init__)
        1    0.000    0.000    0.000    0.000 runpy.py:104(_get_module_details)
        1    0.000    0.000    0.005    0.005 runpy.py:197(run_module)
        1    0.000    0.000    0.004    0.004 runpy.py:64(_run_code)
        1    0.000    0.000    0.000    0.000 util.py:73(find_spec)
```

* same outputs for the profiler implemented in C:

```sh
python -m cProfile -m profile_me
# outputs:
sys.path: ['...test', '/home/luc/.pyenv/versions/3.8.6/lib/python38.zip', '/home/luc/.pyenv/versions/3.8.6/lib/python3.8', '/home/luc/.pyenv/versions/3.8.6/lib/python3.8/lib-dynload', '/home/luc/.pyenv/versions/3.8.6/lib/python3.8/site-packages']
         1268 function calls (1240 primitive calls) in 0.173 seconds

   Ordered by: standard name

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:103(release)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:143(__init__)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:147(__enter__)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:151(__exit__)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:157(_get_module_lock)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:176(cb)
     10/1    0.000    0.000    0.001    0.001 <frozen importlib._bootstrap>:211(_call_with_frames_removed)
       86    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:222(_verbose_message)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:342(__init__)
        2    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:35(_new_module)
        9    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:376(cached)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:389(parent)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:397(has_location)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:477(_init_module_attrs)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:549(module_from_spec)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:58(__init__)
      6/1    0.000    0.000    0.002    0.002 <frozen importlib._bootstrap>:650(_load_unlocked)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:725(find_spec)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:78(acquire)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:800(find_spec)
       21    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:863(__enter__)
       21    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap>:867(__exit__)
        7    0.000    0.000    0.001    0.000 <frozen importlib._bootstrap>:890(_find_spec)
      6/1    0.000    0.000    0.002    0.002 <frozen importlib._bootstrap>:956(_find_and_load_unlocked)
      6/1    0.000    0.000    0.002    0.002 <frozen importlib._bootstrap>:986(_find_and_load)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1010(path_stats)
        4    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1088(__init__)
        4    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1099(create_module)
        4    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1107(exec_module)
       23    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1265(_path_importer_cache)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1302(_get_spec)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1334(find_spec)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1426(_get_spec)
       17    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:1431(find_spec)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:294(cache_from_source)
       17    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:40(_relax_case)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:424(_get_cached)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:456(_check_name_wrapper)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:493(_classify_pyc)
        9    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:51(_unpack_uint32)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:526(_validate_timestamp_pyc)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:578(_compile_bytecode)
       72    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:62(_path_join)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:629(spec_from_file_location)
       72    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:64(<listcomp>)
        6    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:68(_path_split)
        2    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:774(create_module)
      2/1    0.000    0.000    0.002    0.002 <frozen importlib._bootstrap_external>:777(exec_module)
       27    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:80(_path_stat)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:849(get_code)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:90(_path_is_mode_type)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:939(__init__)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:964(get_filename)
        3    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:969(get_data)
        7    0.000    0.000    0.000    0.000 <frozen importlib._bootstrap_external>:99(_path_isfile)
        1    0.000    0.000    0.173    0.173 <string>:1(<module>)
        1    0.000    0.000    0.000    0.000 bisect.py:1(<module>)
        1    0.000    0.000    0.173    0.173 profile_me.py:1(<module>)
        1    0.000    0.000    0.171    0.171 profile_me.py:5(i_do_things)
        1    0.000    0.000    0.001    0.001 random.py:1(<module>)
        1    0.000    0.000    0.000    0.000 random.py:103(__init_subclass__)
        1    0.000    0.000    0.000    0.000 random.py:123(seed)
        5    0.000    0.000    0.000    0.000 random.py:250(_randbelow_with_getrandbits)
        5    0.000    0.000    0.000    0.000 random.py:285(choice)
        1    0.000    0.000    0.000    0.000 random.py:709(SystemRandom)
        1    0.000    0.000    0.000    0.000 random.py:78(Random)
        1    0.000    0.000    0.000    0.000 random.py:94(__init__)
        1    0.000    0.000    0.000    0.000 runpy.py:104(_get_module_details)
        1    0.000    0.000    0.173    0.173 runpy.py:197(run_module)
        1    0.000    0.000    0.173    0.173 runpy.py:64(_run_code)
        1    0.000    0.000    0.000    0.000 util.py:73(find_spec)
        3    0.000    0.000    0.000    0.000 {built-in method _imp._fix_co_filename}
       33    0.000    0.000    0.000    0.000 {built-in method _imp.acquire_lock}
        4    0.000    0.000    0.000    0.000 {built-in method _imp.create_dynamic}
        4    0.000    0.000    0.000    0.000 {built-in method _imp.exec_dynamic}
        7    0.000    0.000    0.000    0.000 {built-in method _imp.is_builtin}
        7    0.000    0.000    0.000    0.000 {built-in method _imp.is_frozen}
       33    0.000    0.000    0.000    0.000 {built-in method _imp.release_lock}
       12    0.000    0.000    0.000    0.000 {built-in method _thread.allocate_lock}
       12    0.000    0.000    0.000    0.000 {built-in method _thread.get_ident}
        2    0.000    0.000    0.000    0.000 {built-in method builtins.__build_class__}
      4/1    0.000    0.000    0.173    0.173 {built-in method builtins.exec}
       36    0.000    0.000    0.000    0.000 {built-in method builtins.getattr}
       30    0.000    0.000    0.000    0.000 {built-in method builtins.hasattr}
       30    0.000    0.000    0.000    0.000 {built-in method builtins.isinstance}
       23    0.000    0.000    0.000    0.000 {built-in method builtins.len}
        1    0.000    0.000    0.000    0.000 {built-in method builtins.print}
        9    0.000    0.000    0.000    0.000 {built-in method from_bytes}
        3    0.000    0.000    0.000    0.000 {built-in method io.open_code}
        3    0.000    0.000    0.000    0.000 {built-in method marshal.loads}
        1    0.000    0.000    0.000    0.000 {built-in method math.exp}
        2    0.000    0.000    0.000    0.000 {built-in method math.log}
        1    0.000    0.000    0.000    0.000 {built-in method math.sqrt}
       13    0.000    0.000    0.000    0.000 {built-in method posix.fspath}
        1    0.000    0.000    0.000    0.000 {built-in method posix.register_at_fork}
       27    0.000    0.000    0.000    0.000 {built-in method posix.stat}
        5    0.171    0.034    0.171    0.034 {built-in method time.sleep}
        1    0.000    0.000    0.000    0.000 {function Random.seed at 0x7f3b4ac22310}
        5    0.000    0.000    0.000    0.000 {method 'bit_length' of 'int' objects}
        1    0.000    0.000    0.000    0.000 {method 'disable' of '_lsprof.Profiler' objects}
       11    0.000    0.000    0.000    0.000 {method 'endswith' of 'str' objects}
       12    0.000    0.000    0.000    0.000 {method 'get' of 'dict' objects}
       15    0.000    0.000    0.000    0.000 {method 'getrandbits' of '_random.Random' objects}
       78    0.000    0.000    0.000    0.000 {method 'join' of 'str' objects}
        6    0.000    0.000    0.000    0.000 {method 'pop' of 'dict' objects}
        3    0.000    0.000    0.000    0.000 {method 'read' of '_io.BufferedReader' objects}
       44    0.000    0.000    0.000    0.000 {method 'rpartition' of 'str' objects}
      150    0.000    0.000    0.000    0.000 {method 'rstrip' of 'str' objects}
        2    0.000    0.000    0.000    0.000 {method 'startswith' of 'str' objects}
        1    0.000    0.000    0.000    0.000 {method 'update' of 'dict' objects}
```

I hope this helps.

**Comment by joerick** [comment]
Thanks! So. Yes this looks like a bug. I have discovered something interesting - if you invoke your script like `python -m pyinstrument -m my_module` it works fine. But `pyinstrument -m my_module` does not work. My unit tests use the former style of invocation. I'm still investigating more...

**Comment by joerick** [comment]
Would you mind trying the version on master branch? You can install it by running `pip install https://github.com/joerick/pyinstrument/archive/master.zip`

**Comment by lucsorel** [comment]
The master version worked :)

```sh
pyinstrument -m profile_me
# outputs:
sys.path: ['test', 'test/.venv/bin', '.../3.8.6/lib/python38.zip', '.../3.8.6/lib/python3.8', '.../3.8.6/lib/python3.8/lib-dynload', 'test/.venv/lib/python3.8/site-packages']

  _     ._   __/__   _ _  _  _ _/_   Recorded: 21:16:21  Samples:  5
 /_//_/// /_\ / //_// / //_'/ //     Duration: 0.102     CPU time: 0.001
/   _/                      v3.2.0

Program: profile_me

0.102 <module>  <string>:1
└─ 0.102 run_module  runpy.py:197
      [2 frames hidden]  runpy
         0.102 _run_code  runpy.py:64
         └─ 0.102 <module>  profile_me.py:1
            └─ 0.102 i_do_things  profile_me.py:5
               └─ 0.102 sleep  <built-in>:0
```

Thank you @joerick for your reactivity, this is really great. Looking forward to installing 3.2.1 soon! ;)

Maybe an additional unit test with the `pyinstrument -m my_module` would secure this fix?

**Comment by joerick** [comment]
> Maybe an additional unit test with the `pyinstrument -m my_module` would secure this fix?

That's already in :) I'll do a release now.

**Comment by joerick** [comment]
Released as v3.3.0
